src/multiplexed-session.ts

```diff
@@ -0,0 +1,255 @@
+ * @param {Database} database The database to create a multiplexed session for.
+ */
+export interface MultiplexedSessionInterface {
+  /**
+   * When called creates a multiplexed session.
+   *
+   * @name MultiplexedSessionInterface#createSession
+   */
+  createSession(): void;
+
+  /**
+   * When called returns a multiplexed session.
+   *
+   * @name MultiplexedSessionInterface#getSession
+   * @param {GetSessionCallback} callback The callback function.
+   */
+  getSession(callback: GetSessionCallback): void;
+}
+
+/**
+ * Class used to manage connections to Spanner using multiplexed session.
+ *
+ * **You don't need to use this class directly, connections will be handled for
+ * you.**
+ *
+ * @class
+ * @extends {EventEmitter}
+ */
+export class MultiplexedSession
+  extends EventEmitter
+  implements MultiplexedSessionInterface
+{
+  database: Database;
+  // frequency to create new mux session
+  refreshRate: number;
+  _multiplexedSession: Session | null;
+  _refreshHandle!: NodeJS.Timer;
+  _observabilityOptions?: ObservabilityOptions;
+  constructor(database: Database) {
+    super();
+    this.database = database;
+    // default frequency is 7 days
+    this.refreshRate = 7;
+    this._multiplexedSession = null;
+    this._observabilityOptions = database._observabilityOptions;
+  }
+
+  /**
+   * Creates a new multiplexed session and manages its maintenance.
+   *
+   * This method initiates the session creation process by calling the `_createSession` method, which returns a Promise.
+   */
+  createSession(): void {
+    this._createSession()
+      .then(() => {
+        this._maintain();
+      })
+      .catch(err => {
+        this.emit('error', err);
+      });
+  }
+
+  /**
+   * Creates a new multiplexed session.
+   *
+   * This method sends a request to the database to create a new session with multiplexing enabled.
+   * The response from the database would be an array, the first value of the array will be containing the multiplexed session.
+   *
+   * @returns {Promise<void>} A Promise that resolves when the session has been successfully created and assigned, an event
+   * `mux-session-available` will be emitted to signal that the session is ready.
+   *
+   * In case of error, an error will get emitted along with the erorr event.
+   *
+   * @private
+   */
+  async _createSession(): Promise<void> {
+    const traceConfig = {
+      opts: this._observabilityOptions,
+      dbName: this.database.formattedName_,
+    };
+    return startTrace(
+      'MultiplexedSession.createSession',
+      traceConfig,
+      async span => {
+        span.addEvent('Requesting a multiplexed session');
+        try {
+          const [createSessionResponse] = await this.database.createSession({
+            multiplexed: true,
+          });
+          this._multiplexedSession = createSessionResponse;
+          span.addEvent(
+            `Created multiplexed session ${this._multiplexedSession.formattedName_}`
```

Comment:
How this `${this._multiplexedSession.formattedName_}` will look like ? I don't think you should ad the name, instead add the `session.id` in span.

Reply:
>>How this ${this._multiplexedSession.formattedName_} will look like ?

it looks like projects/my-project/instances/my-instance/databases/my-database/sessions/my-session

>> How this ${this._multiplexedSession.formattedName_} will look like ? I don't think you should ad the name, instead add the session.id in span.

sure, I have updated the `${this._multiplexedSession.formattedName_}` to `${this._multiplexedSession.id}`